kong/api/routes/certificates.lua

```diff
@@ -3,71 +3,189 @@ local utils = require "kong.tools.utils"
+local function create_certificate(self, dao_factory, helpers)
+  local snis
+  if type(self.params.snis) == "string" then
+    snis = utils.split(self.params.snis, ",")
+    self.params.snis = nil
+  end
+
+  if snis then
+    -- dont add the certificate or any snis if we have an SNI conflict
+    -- its fairly inefficient that we have to loop twice over the datastore
+    -- but no support for OR queries means we gotsta!
+    local snis_in_request = {}
+
+    for _, sni in ipairs(snis) do
+      if snis_in_request[sni] then
+        return helpers.responses.send_HTTP_CONFLICT(
+          "duplicate requested sni name " .. sni
+        )
       end
 
-      if snis then
-        -- dont add the certificate or any snis if we have an SNI conflict
-        -- its fairly inefficient that we have to loop twice over the datastore
-        -- but no support for OR queries means we gotsta!
-        local seen = {}
-
-        for _, sni in ipairs(snis) do
-          if seen[sni] then
-            return helpers.responses.send_HTTP_CONFLICT(
-              "duplicate requested sni name " .. sni
-            )
-          end
-
-          local cnt, err = dao_factory.ssl_servers_names:count({
-            name = sni,
-          })
-          if err then
-            return helpers.yield_error(err)
-          end
-
-          if cnt > 0 then
-            return helpers.responses.send_HTTP_CONFLICT(
-              "entry already exists with name " .. sni
-            )
-          end
-
-          seen[sni] = true
-        end
+      local cnt, err = dao_factory.ssl_servers_names:count({
+        name = sni,
+      })
+      if err then
+        return helpers.yield_error(err)
+      end
+
+      if cnt > 0 then
+        return helpers.responses.send_HTTP_CONFLICT(
+          "entry already exists with name " .. sni
+        )
       end
 
-      local ssl_cert, err = dao_factory.ssl_certificates:insert(self.params)
+      snis_in_request[sni] = true
+    end
+  end
+
+  local ssl_cert, err = dao_factory.ssl_certificates:insert(self.params)
+  if err then
+    return helpers.yield_error(err)
+  end
+
+  -- insert SNIs if given
+
+  if snis then
+    ssl_cert.snis = setmetatable({}, cjson.empty_array_mt)
+
+    for _, sni in ipairs(snis) do
+      local ssl_server_name = {
+        name                = sni,
+        ssl_certificate_id  = ssl_cert.id,
+      }
+
+      local row, err = dao_factory.ssl_servers_names:insert(ssl_server_name)
       if err then
         return helpers.yield_error(err)
       end
 
-      -- insert SNIs if given
+      table.insert(ssl_cert.snis, row.name)
+    end
+  end
 
-      if snis then
-        ssl_cert.snis = {}
+  return helpers.responses.send_HTTP_CREATED(ssl_cert)
+end
 
-        for _, sni in ipairs(snis) do
-          local ssl_server_name = {
-            name                = sni,
-            ssl_certificate_id  = ssl_cert.id,
-          }
+local function update_certificate(self, dao_factory, helpers)
```

Comment:
style: we jump 2 lines between function declarations consistency in this codebase. Would you mind updating? Thanks!